shopinvader_product_variant_selector/tests/test_shopinvader_variant.py

```diff
@@ -0,0 +1,301 @@
+                    ],
+                }
+            ],
+        )
+
+    def test_selector_with_two_dimension(self):
+        self.env.ref("product.product_attribute_line_3").unlink()
+        variant = self._configure_and_get_variant("White-16 GB")
+        self.assertEqual(
+            variant.variant_selector,
+            [
+                {
+                    "name": "Color",
+                    "values": [
+                        {
+                            "name": "White",
+                            "sku": "White-16 GB",
+                            "selected": True,
+                            "available": True,
+                        },
+                        {
+                            "name": "Black",
+                            "sku": "Black-16 GB",
+                            "selected": False,
+                            "available": True,
+                        },
+                    ],
+                },
+                {
+                    "name": "Memory",
+                    "values": [
+                        {
+                            "name": "16 GB",
+                            "sku": "White-16 GB",
+                            "selected": True,
+                            "available": True,
+                        },
+                        {
+                            "name": "32 GB",
+                            "sku": "White-32 GB",
+                            "selected": False,
+                            "available": True,
+                        },
+                    ],
+                },
+            ],
+        )
+
+    def test_selector_with_two_dimension_and_inactive(self):
+        self.env.ref("product.product_attribute_line_3").unlink()
+        variant = self._configure_and_get_variant("White-16 GB")
+        self._inactive_variant(["White-32 GB"])
+        self.assertEqual(
+            variant.variant_selector,
+            [
+                {
+                    "name": "Color",
+                    "values": [
+                        {
+                            "name": "White",
+                            "sku": "White-16 GB",
+                            "selected": True,
+                            "available": True,
+                        },
+                        {
+                            "name": "Black",
+                            "sku": "Black-16 GB",
+                            "selected": False,
+                            "available": True,
+                        },
+                    ],
+                },
+                {
+                    "name": "Memory",
+                    "values": [
+                        {
+                            "name": "16 GB",
+                            "sku": "White-16 GB",
+                            "selected": True,
+                            "available": True,
+                        },
+                        {
+                            "name": "32 GB",
```

Comment:
white 32 GB?

Reply:
The SKU is used to be able to select the right variant on the front. As this value is not available there is not sku (variant is not active so this variant do not exist on elastic/algolia)